FEATURE_GATES.md

```diff
@@ -168,3 +168,89 @@ services: if two HTTPRoute rules use both serviceA and serviceB in their
 backendRefs, KIC will generate a single Kong service with endpoints from both
 serviceA and serviceB, named for the first rule and match indices with that
 combination of Services.
+
+## Using KongServiceFacade
+
+In KIC 3.1.0 we introduced a new feature called `KongServiceFacade`. Currently, we only
+support `KongServiceFacade` to be used as a backend for `netv1.Ingress`es.
+If you find this might be useful to you in other contexts (e.g. Gateway API's `HTTPRoute`s),
+please let us know in the [#5216](https://github.com/Kong/kubernetes-ingress-controller/issues/5216)
+issue tracking this effort.
+
+### Installation
+
+`KongServiceFacade` feature is currently in `Alpha` maturity and is disabled by default.
+To start using it, you must not only enable the feature gate (`KongServiceFacade=true`),
+but also install the `KongServiceFacade` CRD which is distributed in a separate
+package we named `incubator`. You can install it by running:
+
+```shell
+kubectl apply -k github.com/Kong/kubernetes-ingress-controller/main/config/crd/incubator?ref=main
+```
+
+### Usage
+
+Once the CRD is installed and the feature gate is set to `true`, you can start using it by creating
+`KongServiceFacade` objects and use them as your `netv1.Ingress` backends. For example, to create
+a `KongServiceFacade` that points to a service named `my-service` in the `default` namespace and uses
+its port number `80`, you can run:
+
+```shell
+kubectl apply -f - <<EOF
+apiVersion: incubator.konghq.com/v1alpha1
+kind: KongServiceFacade
+metadata:
+  name: my-service-facade
+  namespace: default
+  annotations:
+    kubernetes.io/ingress.class: kong
+spec:
+  backendRef:
+    name: my-service
+    port: 80
+EOF
+```
+
+For a complete CRD reference please see the [incubator-crd-reference] document. 
+
+To use the `KongServiceFacade` as a backend for your `netv1.Ingress`, you can create an `Ingress`
+like the following:
+
+```shell
+kubectl apply -f - <<EOF
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: my-ingress
+  namespace: default
+  annotations:
+    konghq.com/strip-path: "true"
+spec:
+  ingressClassName: kong
+  rules:
+  - http:
+      paths:
+      - path: /my-service
+        pathType: Prefix
+        backend:
+          resource:
+            apiGroup: incubator.konghq.com
```

Comment:
This is a real URL that redirects to docs. Do we have a naming scheme for api versions with KIC/for k8s at Kong in general?

Reply:
> This is a real URL that redirects to docs.
As long as we are in control (Kong) of the domain, URL being real shouldn't be a big deal. No one expects API groups to be valid/invalid URLs but unique.

>Do we have a naming scheme for api versions with KIC/for k8s at Kong in general?
In Gateway Operator we're using `gateway-operator.konghq.com`. In KIC we've been using `configuration.konghq.com` only so far. Kuma uses `kuma.io` domain. 

We cannot fix `configuration.konghq.com` to be KIC-specific enough, but if you like we can make `incubator` to be `incubator.ingresscontroller.konghq.com` so we're sure we're in a KIC namespace.

What I don't like about it is that we'll be inconsistent by doing so.

Reply:
After the Slack discussion: https://github.com/Kong/kubernetes-ingress-controller/pull/5302

Reply:
Waiting for [#968](https://github.com/Kong/charts/pull/968), [#969](https://github.com/Kong/charts/pull/969) in charts to be merged to resolve this discussion.